src/eda_server/api/activation.py

```diff
@@ -26,6 +26,39 @@
 router = APIRouter(tags=["activations"])
 
 
+async def dependent_object_exists_or_exception(db: AsyncSession, activation):
+    activation_dependent_objects = [
+        (models.rulebooks, "rulebook"),
+        (models.inventories, "inventory"),
+        (models.extra_vars, "extra_var"),
+        (models.projects, "project"),
+    ]
+
+    dep_object_ids = [
+        activation.rulebook_id,
+        activation.inventory_id,
+        activation.extra_var_id,
+        activation.project_id,
+    ]
+
+    for dep_object, object_id in zip(
+        activation_dependent_objects, dep_object_ids
+    ):
+        dependent_object = (
+            await db.execute(
+                sa.select(dep_object[0]).where(dep_object[0].c.id == object_id)
```

Comment:
No need to select entire record to check its existence. Use `SELECT EXISTS` instead.